data/wild_hands.py

```diff
@@ -26,8 +39,35 @@ def __getitem__(self, idx):
         Returns:
             tuple: (image, target) where target is the same image.
         """
-        pass
+        img = Image.open(self.img_dirs[idx]).convert("RGB")
+        img = self.transform(img)
+
+        target = Image.open(self.label_dirs[idx]).convert("RGB")  # not sure mn el convert tbh
+        target = self.transform(target)
+        return img, target
+
+    def get_img_dirs(self):
+        imgs = []
+        labels = []
+        for dir_path, sub_dirs, files in os.walk(self.dataset_path):
+            for file in files:
+                if 'ask' in file:
+                    labels.append(os.path.join(dir_path, file))
+                else:
```

Comment:
An else statement can be unsafe in some conditions, Check also if the file has an image format you know (except ```png```).

Reply:
the img frames also end with .png

Reply:
I rechecked the dataset and found you were correct, check if it's ```png``` also.

Reply:
now I check if the extension is either jpg or png for both the labels and the images

Reply:
because level 3 images are jpg

Reply:
Right